### ISSUE TITLE
'./configure --enable-shared': build error on Linux when using Wayland

### ISSUE THREAD

**Issue by Albrecht-S** [fltk/fltk#872]
**Description**
Shared libs can't be built with configure/make when using Wayland on Linux.

**To Reproduce**
```
$ make distclean; autoconf; ./configure --enable-shared
$ make
```

Note: works well if Wayland is disabled `./configure --enable-shared --disable-wayland`.

**FLTK Version**
 - Version: 1.4 Git master, commit 5df5daf78e135e0c8371012ffe56df5cae357755

**FLTK Configure / Build Options**
See above.

**Operating System / Platform:**
 - OS: Linux
 - OS Version: Debian 12 (Stable, Bookworm)

**Linux/Unix Runtime, if applicable:**
 - Wayland

**Additional context**
- I checked out commit 102437f1eedce8589d947bf8791d6d5f413d445b (100 commits back, Nov 26, 2023) and this one failed as well.
- I **assume** that building shared libs with _configure/make **and** Wayland_ never worked (but this may be wrong)
- Building shared libs with CMake and Wayland works fine.

**Comment by Albrecht-S** [comment]
@ManoloFLTK I assigned this issue to you because I don't know how to fix this. Prio 3 (Moderate) because it's only relevant for building shared libs with configure/make.

Build error messages:
```
=== making src ===
make[1]: Entering directory '/git/fltk/master/src'
g++ -L. -pthread -lpthread -lm -lwayland-cursor -lwayland-client -lxkbcommon -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo -lX11 -ldl -Wl,-no-undefined -Wl,-Bsymbolic-functions -Wl,-gc-sections -Wl,-soname,libfltk_gl.so.1.4 -pthread -lpthread -lm -lwayland-cursor -lwayland-client -lxkbcommon -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo -lX11 -ldl -ldbus-1 -lgtk-3 -lgdk-3 -lz -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo -lXinerama -lXcursor -lXfixes -lXrender -lX11 -shared -fPIC -o libfltk_gl.so.1.4 ...
/usr/bin/ld: drivers/Wayland/Fl_Wayland_Gl_Window_Driver.o: in function `Fl_Wayland_Gl_Window_Driver::pixels_per_unit()':
Fl_Wayland_Gl_Window_Driver.cxx:(.text._ZN27Fl_Wayland_Gl_Window_Driver15pixels_per_unitEv+0x2f): undefined reference to `Fl_Wayland_Window_Driver::wld_scale()'
/usr/bin/ld: drivers/Wayland/Fl_Wayland_Gl_Window_Driver.o: in function `Fl_Wayland_Gl_Window_Driver::resize(int, int, int)':
Fl_Wayland_Gl_Window_Driver.cxx:(.text._ZN27Fl_Wayland_Gl_Window_Driver6resizeEiii+0x3d): undefined reference to `Fl_Wayland_Window_Driver::wld_scale()'
/usr/bin/ld: drivers/Wayland/Fl_Wayland_Gl_Window_Driver.o: in function `Fl_Wayland_Gl_Window_Driver::alpha_mask_for_string(char const*, int, int, int, int)':
Fl_Wayland_Gl_Window_Driver.cxx:(.text._ZN27Fl_Wayland_Gl_Window_Driver21alpha_mask_for_stringEPKciiii+0xeb): undefined reference to `Fl_Wayland_Graphics_Driver::offscreen_buffer(unsigned long)'
/usr/bin/ld: drivers/Wayland/Fl_Wayland_Gl_Window_Driver.o: in function `Fl_Wayland_Gl_Window_Driver::make_current_before()':
Fl_Wayland_Gl_Window_Driver.cxx:(.text._ZN27Fl_Wayland_Gl_Window_Driver19make_current_beforeEv+0x4d): undefined reference to `Fl_Wayland_Window_Driver::wld_scale()'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:532: libfltk_gl.so.1.4] Error 1
make[1]: Leaving directory '/git/fltk/master/src'
```

**Comment by Albrecht-S** [comment]
Thanks, the build works now.

**Comment by ManoloFLTK** [comment]
@Albrecht-S Do you have any idea why CMake-build doesn't need FL_EXPORT in classes `Fl_Wayland_Graphics_Driver` and `Fl_Wayland_Window_Driver` ? The configure-based build procedure needs to access member functions of these classes which lie in `libfltk.so` to produce `libfltk_gl.so`, and therefore requires these classes to be exported.

**Comment by Albrecht-S** [comment]
> Do you have any idea why CMake-build doesn't need FL_EXPORT in classes Fl_Wayland_Graphics_Driver and Fl_Wayland_Window_Driver ?

I was also wondering why it had worked in CMake builds if the FL_EXPORT was missing. I **assume** (and I found a hint in the CMake docs) that CMake exports all symbols by default using a linker flag like `--export-all-symbols` or something like that. Use your favorite search engine and search for `--export-all-symbols` and you'll see some hints. I intend to dig deeper into this while I'm working on my CMake updates.

---------------

PS: OT, just for your info: I'm currently working on an issue that `FL_EXPORT bool fl_disable_wayland = true;` does no longer work as documented if programs are built with CMake which is a similar symbol export problem. Don't worry, I found the cause (it's a new "feature" since CMake 3.4) and I'm working on a fix. I will commit this soon. It needs some specific documentation for user programs that are affected too if built with `CMake >= 3.4` (minimum version).
